### PR TITLE
Align deployment of Safe to be of same version as Safe UI-deployed and having fallback handler attached

### LINKED ISSUE #1361: Fractal UI deploys out-of-spec Safes
- Upgrade the actual Safe contract version from 1.3.0 to 1.3.0+L2.
- Make sure the Safe contract is deployed with a CompatibilityFallbackHandler contract attached to it.

### PR DESCRIPTION
**Pull request by mudrila**
## Description
This PR swaps all the `GnosisSafe` usages to `GnosisSafeL2` and adds `fallbackHandler` to the create Safe flow, right into `setup` function call.

## Issue

Closes #1361 

## Testing
1. Deploy brand new Safe via Fractal
2. Create new Safe via Safe UI, go to Safe's Settings page
3. Go to Safe UI and find Fractal-created Safe, go to Safe's Settings page
4. Compare Safe created via Fractal and Safe created via Safe UI - their version should be exactly the same (`1.3.0+L2`) and attached Fallback Handler (could be found under "Modules" tab) also should be exactly the same. See screenshots below for references on how it should look like

## Screenshots (if applicable)
Знімок екрана 2024-02-26 о 19 06 43
Знімок екрана 2024-02-26 о 19 43 23

### CODE REVIEW

File: src/providers/NetworkConfig/networks/mainnet.ts [hunk 1]
```diff
@@ -47,7 +48,11 @@ export const mainnetConfig: NetworkConfig = {
     erc20FreezeVotingMasterCopy: ERC20FreezeVoting.address,
     erc721FreezeVotingMasterCopy: '', // TODO - Add actual address once contract is deployed on mainnet
     multisigFreezeGuardMasterCopy: MultisigFreezeGuard.address,
-    safe: getSafeSingletonDeployment({ version: SAFE_VERSION, network: CHAIN_ID.toString() })
+    fallbackHandler: getCompatibilityFallbackHandlerDeployment({
+      version: SAFE_VERSION,
+      network: CHAIN_ID.toString(),
+    })?.defaultAddress!,
```
Comment (adamgall) [inline src/providers/NetworkConfig/networks/mainnet.ts:54]: I don't think we should be using the `defaultAddress` here... that's why you're seeing the "wrong address" for Sepolia ("11155111"). See: https://github.com/safe-global/safe-deployments/blob/main/src/assets/v1.3.0/compatibility_fallback_handler.json

I think you need to dig into the `networkAddresses` object to pull out the proper address based on the network id.

**Review by Da-Colon — APPROVED** [review]
Surprised we still have to rely on the typechain files for these. But tested and it indeed works. I see that the Contract+L2 is correctly displayed in the settings and the CompatibilityFallbackHandler has been set up as a module.

File: src/providers/NetworkConfig/networks/mainnet.ts [hunk 2]
```diff
@@ -47,7 +48,11 @@ export const mainnetConfig: NetworkConfig = {
...
+    safe: getSafeL2SingletonDeployment({ version: SAFE_VERSION, network: CHAIN_ID.toString() })
       ?.defaultAddress!,
```
Comment (adamgall) [inline src/providers/NetworkConfig/networks/mainnet.ts:56]: Same issue here. Don't rely on this `defaultAddress`, but dig into the `networkAddresses` object to find the correct address.

File: src/providers/NetworkConfig/networks/polygon.ts [hunk 3]
```diff
@@ -47,7 +48,11 @@ export const polygonConfig: NetworkConfig = {
     erc20FreezeVotingMasterCopy: ERC20FreezeVoting.address,
     erc721FreezeVotingMasterCopy: '',
     multisigFreezeGuardMasterCopy: MultisigFreezeGuard.address,
-    safe: getSafeSingletonDeployment({ version: SAFE_VERSION, network: CHAIN_ID.toString() })
+    fallbackHandler: getCompatibilityFallbackHandlerDeployment({
+      version: SAFE_VERSION,
+      network: CHAIN_ID.toString(),
+    })?.networkAddresses[CHAIN_ID.toString()]!,
```
Comment (adamgall) [inline src/providers/NetworkConfig/networks/polygon.ts:54]: Correct!

File: src/providers/NetworkConfig/networks/polygon.ts [hunk 4]
```diff
@@ -47,7 +48,11 @@ export const polygonConfig: NetworkConfig = {
...
+    safe: getSafeL2SingletonDeployment({ version: SAFE_VERSION, network: CHAIN_ID.toString() })
       ?.defaultAddress!,
```
Comment (adamgall) [inline src/providers/NetworkConfig/networks/polygon.ts:56]: Fix please!

File: src/providers/NetworkConfig/networks/sepolia.ts [hunk 5]
```diff
@@ -48,7 +49,11 @@ export const sepoliaConfig: NetworkConfig = {
     erc20FreezeVotingMasterCopy: ERC20FreezeVoting.address,
     erc721FreezeVotingMasterCopy: ERC721FreezeVoting.address,
     multisigFreezeGuardMasterCopy: MultisigFreezeGuard.address,
-    safe: getSafeSingletonDeployment({ version: SAFE_VERSION, network: CHAIN_ID.toString() })
+    fallbackHandler: getCompatibilityFallbackHandlerDeployment({
+      version: SAFE_VERSION,
+      network: CHAIN_ID.toString(),
+    })?.networkAddresses[CHAIN_ID.toString()]!,
```
Comment (adamgall) [inline src/providers/NetworkConfig/networks/sepolia.ts:55]: Correct!

File: src/providers/NetworkConfig/networks/sepolia.ts [hunk 6]
```diff
@@ -48,7 +49,11 @@ export const sepoliaConfig: NetworkConfig = {
...
+    safe: getSafeL2SingletonDeployment({ version: SAFE_VERSION, network: CHAIN_ID.toString() })
       ?.defaultAddress!,
```
Comment (adamgall) [inline src/providers/NetworkConfig/networks/sepolia.ts:57]: Fix please!

**Review by adamgall — CHANGES_REQUESTED** [review]
Whenever using the `safe-deployments` package to `get` singleton addresses, the filter simply confirms that the contract has been deployed on the given network.

It's still up to the caller to dig into the `networkAddresses` object of the result to find the _official_ address for a given chain, do not rely on the `defaultAddress`!

This applies to the new `CompatibilityFallbackHandler` code (which seems good for Sepolia and Polygon, but might as well do it for Mainnet too), as well as the actual Master Safe Singleton deployment address as well (for ALL THREE networks).

**Comment by adamgall** [comment]
> Surprised we still have to rely on the typechain files for these. But tested and it indeed works. I see that the Contract+L2 is correctly displayed in the settings and the CompatibilityFallbackHandler has been set up as a module.

Pretty lucky that we had those `GnosisSafeL2` types in that `usul` folder, huh! It looks like the only reason we had those types in the project, before, this PR, was to get the Multisend typing.

**Review by adamgall — APPROVED** [review]
LGTM!

**Pull request merged**